bpget/bp_get.go

```diff
@@ -56,18 +56,43 @@ func GetBlueprint(service string, version string) []*BluePrint {
 	return l
 }
 
-func getFileName(s string, v string) string {
+func getFileName(s string, v string, l string) string {
 	t := time.Now().Format("2006_01_02_15_04_05")
-	fileName := strings.Join([]string{s, v, t}, "_")
+	var fileName string
+	if l != "" {
+		fileName = l + strings.Join([]string{s, v, t}, "_")
+	} else {
+		fileName = strings.Join([]string{s, v, t}, "_")
+	}
+
 	return fileName + ".csv"
 }
 
 func main() {
 	servicePtr := flag.String("service", "nova", "The service name of OpenStack (nova, cinder...)")
 	versionPtr := flag.String("version", "queens", "The version of service (pike, queens...)")
+	locationPtr := flag.String("location", "", "The location of the output file.")
 	flag.Parse()
 	service := *servicePtr
 	version := *versionPtr
+	location := *locationPtr
+	_, err := os.Stat(location)
+	if os.IsNotExist(err) {
+		folders := strings.Split(location, "\\")
+		folder := folders[0]
+		var deep int
+		folderLength := len(folders)
+		if folders[folderLength-1] == "\\" {
+			deep = folderLength
+		} else {
+			deep = folderLength -1
+		}
+
+		for i := 1; i < deep; i++ {
+			folder = folder + "\\" + folders[i]
+			os.Mkdir(folder, os.ModeDir)
```

Comment:
try to use [os.MkdirAll](https://golang.org/pkg/os/#MkdirAll), it seems more convenient to create directory recursively, and also more friendly to mkdir in different OS.